source/patterns/@aws-solutions-constructs/aws-lambda-sagemakerendpoint/README.md

```diff
@@ -0,0 +1,123 @@
+| vpcProps?                     | `ec2.VpcProps`                                                                                                                           | Optional user-provided properties to override the default properties for the new VPC. `enableDnsHostnames`, `enableDnsSupport`, `natGateways` and `subnetConfiguration` are set by the Construct, so any values for those properties supplied here will be overrriden. If `deployVpc?` is not `true` then this property will be ignored.                                                                                                                                                                                                                                                                                                                                              |
+| deployVpc?                    | `boolean`                                                                                                                                | Whether to create a new VPC based on defaults properties, or user-provided `vpcProps`, into which to deploy this construct. Setting this to `true` will deploy the minimal, most private VPC to run the construct. By default, the resources will be deployed in isolated subnets. If the `deployNatGateWay?` is set to `true`, resources will be deployed in private subnets with two NatGateWays deployed in the public subnets. One isolated, or private, subnet in each Availability Zone used by the CDK program. `enableDnsHostnames` and `enableDnsSupport` will both be set to true. If this property is `true` then `existingVpc?` cannot be specified. Defualts to `false`. |  |
+| role?                         | `iam.Role`                                                                                                                               | Optional IAM role, with all required permissions, to be assumed by Sagemaker Service to create resources. The pattern creates an internal role with minimum required permissions. However, the customer can provided a custom role. If a role is provided, the `executionRoleArn` in [`sagemaker.CfnModelProps`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-sagemaker.CfnModelProps.html) must be also specified.                                                                                                                                                                                                                                                   |
```

Comment:
Bigger question - we're asking the client to provide the same information twice. We specify the role here and the role arn in modelProps. Can't we drop this and get the role using fromResource on the modelProps arn? One of things we like to avoid is pulling out individual properties from CDK defined props and promoting them to the construct props. (but I like not having to create the role in advance :-) )

Reply:
I'll remove the `Role` argument and use `iam.Role.fromRoleArn` to import customer's role using `executionRoleArn` provided in `modelProps`.